R (#498)
Codecademy/docs: content/r/concepts/comments/comments.md

---
Title: 'Comments'

> Description: 'A comment is a piece of text within a program that is not executed. It can be used to provide additional information to aid in understanding the code.'

Reviewer: I think this definition can be expanded a bit. Comments (commenting) can serve multiple purposes in addition to clarifying code, such as commenting out a function or section of code during debugging

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Comments'
  - 'Documentation'
CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

> A comment is a piece of text within a program that is not executed. It can be used to provide additional information to aid in understanding the code.
> Many languages support single line and multi-line comments. `R` only supports single-line comments.

Reviewer 2: ```suggestion
A comment is a piece of text within an R program that is not executed. It can be used to provide additional information to aid in understanding the code. Many languages support single line and multi-line comments. Only single-line comments are supported in R.
```

> Comments in `R` begin with a `#` character. Everything after the `#` on that line will be ignored.
> The comment can go at the beginning of the line, or after code that will be executed:

Reviewer 2: ```suggestion
## Syntax

Comments in `R` begin with a `#` character. Everything after the `#` on that line will be ignored. The comment can go at the beginning of the line, or after code that will be executed:
```

> A comment is a piece of text within an R program that is not executed. It can be used to provide additional information to aid in understanding the code. It can also serve to "comment out" blocks of code that the programmer doesn't want to be executed at the moment but for some reason, such as archival purposes or debugging, doesn't want to delete from the source code entirely. Many languages support single line and multi-line comments. Only single-line comments are supported in R.

Reviewer: ```suggestion
A comment is a piece of text within an R program that is not executed. It can be used to provide additional information to aid in understanding the code. It can also serve to "comment out" blocks of code that the programmer doesn't want executed, such as for archival purposes or debugging. Many languages support single line and multi-line comments. Only single-line comments are supported in R.
```
